api/assessment/serializers.py

```
@@ -117,18 +115,13 @@ class Meta:
 
     def create(self, validated_data):
         if self.initial_data.get("automate") is True:
-            barrier = Barrier.objects.get(pk=validated_data["barrier_id"])
-            assessment_calculator = AssessmentCalculator(cache=cache)
-            commodity_codes = [c.trimmed_code for c in barrier.commodities.all()]
             try:
-                data = assessment_calculator.calculate(
-                    commodity_codes=commodity_codes,
-                    product=barrier.product,
-                    country1=barrier.country_name,
+                data = calculate_barrier_economic_assessment(
+                    validated_data["barrier_id"]
                 )
+                validated_data["automated_analysis_data"] = data
             except ComtradeError as e:
                 raise ValidationError(e)
-            validated_data["automated_analysis_data"] = data
```

Why has this dictionary assignment been moved inside the try block? It will only hide errors.

---

data is only assigned to inside the block, if data fails to be instantiated an exception is already being raised, thought it might make sense to keep the assignment to validated_data is the same block